common/list.c

```diff
@@ -77,6 +77,15 @@ int list_seq_find(list_t *list, int compare(const void *item, const void *data),
 	return -1;
 }
 
+int list_find(list_t *list, void *item) {
+	for (int i = 0; i < list->length; i++) {
+		if (list->items[i] == item) {
+			return i;
+		}
+	}
+	return -1;
+}
```

Comment:
I think this should just be a static function in that one file.

Reply:
This function can be used in several other places. My plan is to submit a PR after this one which changes these:

* https://github.com/swaywm/sway/blob/e474d87e42d625eb4e0a41a0559bbf0e4ec5ecc5/sway/tree/root.c#L88
* https://github.com/swaywm/sway/blob/e474d87e42d625eb4e0a41a0559bbf0e4ec5ecc5/sway/tree/workspace.c#L296
* https://github.com/swaywm/sway/blob/e474d87e42d625eb4e0a41a0559bbf0e4ec5ecc5/sway/tree/workspace.c#L334
* https://github.com/swaywm/sway/blob/e474d87e42d625eb4e0a41a0559bbf0e4ec5ecc5/sway/tree/workspace.c#L343
* https://github.com/swaywm/sway/blob/e474d87e42d625eb4e0a41a0559bbf0e4ec5ecc5/sway/tree/layout.c#L25
* https://github.com/swaywm/sway/blob/e474d87e42d625eb4e0a41a0559bbf0e4ec5ecc5/sway/tree/layout.c#L129